lib/pam/pam.go

```diff
@@ -280,6 +283,20 @@ func Open(config *Config) (*PAM, error) {
 		return nil, p.codeToError(p.retval)
 	}
 
+	for k, v := range config.Env {
+		// Set a regular OS env var on this process which should be available
+		// to child PAM processes.
+		os.Setenv(k, v)
+
+		// Also set it via PAM-specific pam_putenv, which is respected by
+		// pam_exec (and possibly others), where parent env vars are not.
+		kv := C.CString(fmt.Sprintf("%s=%s", k, v))
```

Comment:
`C.CString` does a malloc under the hood. Will PAM free this memory? If not you'll need to call `C.free` yourself.

Reply:
Ah, nice catch, added `C.free`.
First time using cgo, appreciate all the advice!